build-support/bin/generate_github_workflows.py

```diff
@@ -349,6 +353,7 @@ def test_workflow_jobs(primary_python_version: str, *, cron: bool) -> Jobs:
             "needs": "bootstrap_pants_macos",
             "strategy": {"matrix": {"python-version": [primary_python_version]}},
             "env": MACOS_ENV,
+            "timeout-minutes": 40,
```

Comment:
This one should probably be 20 overall. It looks like you set 20 on the specific step, but also 40 on the overall job. Should probably remove the step timeout.